File: RobotBehavior/Arbiter/Component/Led/LedBlink.java
package de.hska.lat.behavior.arbiter.component.led;

import de.hska.lat.behavior.arbiter.action.ArbiterAction;

public class LedBlink extends ArbiterAction
{

	
	protected float frequency;


	private static final String name = "blink led";
	
	
	public LedBlink(int priority)
	{
		super(priority);
	}

	@Override
	public String getName()
	{
		return(LedBlink.name);
	}


	@Override
	public String toString ()
	{
		String returnString;
		
		returnString = this.getName();
		
		return(returnString);
	}
	
	
	public float getFrequency()
	{
		return frequency;
	}

	public void setFrequency(float frequency)
	{
		this.frequency = frequency;
	}

}

File: RobotBehavior/Arbiter/Component/Led/LedBlink.py
from RobotBehavior.Arbiter.Action.ArbiterAction import ArbiterAction


class LedBlink(ArbiterAction):
	def __init__(self, arbiter_config):
		self._frequency = 0
		arbiter_config['name'] = "Action blink LED"
		super()._init__(arbiter_config)

	
	def get_frequency(self):
		return self._frequency
	

	def set_frequency(self, frequency):
		self._frequency = frequency





